app/controllers/user_actions_controller.rb

```diff
@@ -2,8 +2,7 @@
 
 class UserActionsController < ApplicationController
   def index
-    params.require(:username)
-    params.permit(:filter, :offset, :acting_username, :limit)
+    params.transform_values!(&:to_s).require(:username)
```

Comment:
What we could do on a larger scale is to have a concept of contracts (or similar) to validate parameters before doing anything with their values. We could use gems like [reform](https://github.com/trailblazer/reform) or [dry-validation](https://github.com/dry-rb/dry-validation) for example.